compare.h

```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+//
+template <class T>
+bool compare(const T &a, const T &b) {
+    return a.getCoef() < b.getCoef();
+}
+template <class T>
+T& min(const T &a, const T &b) {
+    if (a.getCoef() < b.getCoef() ) return a;
+    return b;
+}

```

Comment:
Am facut un template mai mult teoretic ca sa il introduc in proiect, nu am reusit sa il leg de ceva astfel incat sa poata fi folosit de mai multe clase. In schimb ar putea fi o functie template pentru comparare daca ar fi implementate mai multe tipuri de connection pool ( poate in loc de festival, un simplu concert, sau pentru a crea grupuri de artisti sau trupe cu numar restrans de locuri ), fiecare avand implementata metoda 'getCoef' care sa returneze un coeficient particular dupa care se realizeaza sortarea.